src/gui/downloadfromurldialog.cpp

```diff
@@ -49,8 +49,12 @@ namespace
             || str.startsWith("https://", Qt::CaseInsensitive)
             || str.startsWith("ftp://", Qt::CaseInsensitive)
             || str.startsWith("magnet:", Qt::CaseInsensitive)
-            || ((str.size() == 40) && !str.contains(QRegularExpression("[^0-9A-Fa-f]")))
-            || ((str.size() == 32) && !str.contains(QRegularExpression("[^2-7A-Za-z]"))));
+            || ((str.size() == 40) && !str.contains(QRegularExpression("[^0-9A-Fa-f]"))) // v1 hex-encoded SHA-1 info-hash
+#if (LIBTORRENT_VERSION_NUM >= 20000)
+            || ((str.size() == 64) && !str.contains(QRegularExpression("[^0-9A-Fa-f]"))) // v2 hex-encoded SHA-256 info-hash
+#endif
+            || ((str.size() == 32) && !str.contains(QRegularExpression("[^2-7A-Za-z]")))); // v1 Base32 encoded SHA-1 info-hash
```

Comment:
Just a question: in what case does this come up? I have never seen v1 Base32 in the wild. Is it officially supported by the BitTorrent spec or libtorrent?

Reply:
I read somewhere, vuze wiki i think, that it predates the hex form. I've never seen one either.

Reply:
@thalieht 

Thanks for the tip, I dug further into this and am now satisfied with the answers I found.

Looks like libtorrent does still officially support it: https://github.com/arvidn/libtorrent/blob/ce367476e29cd2987bfd30c828328a8f72185bf1/src/magnet_uri.cpp#L263

Apparently, mangnets used to be base32-encoded in the past. They eventually transitioned to being generated in hex encoding by default, but support for reading both variants was never removed due to backwards compatibility: https://stackoverflow.com/a/29347969 (authoritative SO answer by Arvid himself).

The Vuze wiki still refers to the 40-byte hex encoding as the "newer" encoding style, instead of the other way around.

TL;DR: remnants of a distant past, but no reason to remove it.